tc/core/polyhedral/cuda/codegen.cc

```diff
@@ -584,12 +592,20 @@ void emitMappedTensorAccess(
 
   // Not promoted, emitting just the mapped subscript.
   if (!promotionInfo.groupId) {
+    auto ctx = context.scop().domain().get_ctx();
+    bool readOnly = context.readOnlySet.count(isl::id(ctx, name)) > 0;
+    if (readOnly) {
+      context.ss << "__ldg(&";
+    }
     context.ss << name;
     for (auto e : subscripts) {
       context.ss << "[";
       emitHalideExpr(e, context);
       context.ss << "]";
     }
+    if (readOnly) {
+      context.ss << ")";
+    }
```

Comment:
Hmm... could we avoid the code duplication by having a `CodegenStatementContext` method that takes an id and an access string and that wraps it in `__ldg` when appropriate?